### PR TITLE
Add error when MinimumOSPlatform is higher than TargetPlatformVersion

### PR DESCRIPTION
**Pull request by wli3**

### CODE REVIEW

**Review by sfoslund — APPROVED** [review]

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets [hunk 1]
```diff
@@ -181,6 +181,13 @@ Copyright (c) .NET Foundation. All rights reserved.
                  ResourceName="NETFrameworkWithoutUsingNETSdkDefaults" />
   </Target>
 
+  <Target Name="_CheckForMinimumOSPlatformHigherThanTargetPlatformVersion" BeforeTargets="_CheckForInvalidConfigurationAndPlatform;Restore;CollectPackageReferences" >
+
+    <NETSdkError Condition="$([MSBuild]::VersionGreaterThan($(MinimumOSPlatform), $(TargetPlatformVersion)))"
```
Comment (sfoslund) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets:186]: We'll need to check that MinimumOSPlatform isn't empty before this

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets [hunk 2]
```diff
@@ -181,6 +181,13 @@ Copyright (c) .NET Foundation. All rights reserved.
                  ResourceName="NETFrameworkWithoutUsingNETSdkDefaults" />
   </Target>
 
+  <Target Name="_CheckForMinimumOSPlatformHigherThanTargetPlatformVersion" BeforeTargets="_CheckForInvalidConfigurationAndPlatform;Restore;CollectPackageReferences" >
```
Comment (dsplaisted) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets:184]: Do we need to run before Restore and CollectPackageReferences?  I don't think so.

**Review by dsplaisted — APPROVED** [review]

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets (hunk 1 above)
Comment (wli3) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets]: Added check for not empty

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets (hunk 2 above)
Comment (wli3) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets]: Removed BeforeTargets - Restore, CollectPackageReferences

**Pull request merged**